editor/src/components/inspector/controls/advanced-grid-modal.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react'
+import { InspectorModal } from '../widgets/inspector-modal'
+import { FlexColumn, H1, SquareButton, UtopiaStyles } from '../../../uuiui'
+import { UIGridRow } from '../widgets/ui-grid-row'
+import type { OptionChainOption } from './option-chain-control'
+import { OptionChainControl } from './option-chain-control'
+import { PrettyLabel } from '../sections/layout-section/flex-container-subsection/flex-container-controls'
+import type { InspectorInfo } from '../common/property-path-hooks'
+import { useInspectorLayoutInfo } from '../common/property-path-hooks'
+import {
+  RadixSelect,
+  regularRadixSelectOption,
+  separatorRadixSelectOption,
+} from '../../../uuiui/radix-components'
+import { optionalMap } from '../../../core/shared/optional-utils'
+import type { FlexAlignment } from 'utopia-api/core'
+import { FlexJustifyContent } from 'utopia-api/core'
+
+export interface AdvancedGridModalProps {
```

Comment:
it seems to me this needs some cleaning up now, for example `key` is not needed, `popupOpen`/`popupClose` don't need to be optional etc.